app/models/merchant.rb

```diff
@@ -1,4 +1,20 @@
 class Merchant < ApplicationRecord
   has_many :items
   
+  def enabled_items
+    items.where(enabled: true)
+  end
+
+  def disabled_items
+    items.where(enabled: false)
+  end
+
+  def top_items
```

Comment:
Could potentially be resolved by using the merchant ID in the where clause: 
``` 
def top_items
  id = self.id
  Item.joins(invoices: :transactions)
      .where(["transactions.result = ? and items.merchant_id = ?", 0, id])
      .select(:name, :id, "sum(invoice_items.unit_price * invoice_items.quantity) as revenue")
      .group(:id)
      .order(revenue: :desc)
      .limit(5)
  end

Reply:
- Thanks for calling out it gets the top items from all merchants.  My thought for it being in the merchant class is that we can run the method on a merchant object and return the top 5 items for that merchant.

-  I implemented your code suggestion and it looks like it works.  

- Question, in your .where, what would be the benefit of using that ? syntax rather than something like  `.where("result = 0 and items.merchant_id = #{id}")` ? Both pieces of code seemed to work.

Reply:
From what I've read it prevents SQL injection attacks. I don't think that is really relevant in this case, but I've just been getting in the habit of passing variables to a where clause through an array. https://api.rubyonrails.org/v7.0.4/classes/ActiveRecord/QueryMethods.html#method-i-where